### PR TITLE
spacegun: init at 0.3.3

### PR DESCRIPTION
**Pull request by kampka**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by bbigras** [comment]
It builds. All 3 binaires run. I didn't test anything else since I don't use kubernetes.

**Comment by kampka** [comment]
@GrahamcOfBorg build spacegun

**Comment by nh2** [comment]
I cannot test this myself, but merging on the base that @bbigras has reviewed it and that @kampka uses the same approach here as in his package `matrix-appservice-slack`.

**Comment by kampka** [comment]
@nh2 @bbigras Thanks for reviewing!

File: pkgs/applications/networking/cluster/spacegun/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+in
+nodePackages."${packageName}".override {
+  nativeBuildInputs = [ pkgs.makeWrapper ];
+
+  postInstall = ''
+    # Patch shebangs in node_modules, otherwise the webpack build fails with interpreter problems
+    patchShebangs --build "$out/lib/node_modules/spacegun/node_modules/"
```
Comment (mweinelt) [inline pkgs/applications/networking/cluster/spacegun/default.nix:17]: This fixes errors like this:

```
sh: /nix/store/zay52n90397hcl16g4fp6dahiaiy9p0s-node_mini-media-player-v1.11.0/lib/node_modules/mini-media-player/node_modules/.bin/webpack: /usr/bin/env: bad interpreter: No such file or directory
```

I hope this improves the discoverability (searchengine) of this fix.

**Pull request merged**